update/gh.go

```diff
@@ -6,70 +6,263 @@ import (
+		expectedChecksum = checksums[d.fullAssetName]
+	}
+	// verify integrity using checksum
+	if expectedChecksum != "" {
+		gotChecksumbytes := sha256.Sum256(buff.Bytes())
+		gotchecksum := hex.EncodeToString(gotChecksumbytes[:])
+		if expectedChecksum != gotchecksum {
+			return nil, errorutil.NewWithTag("checksum", "asset file corrupted: checksum mismatch expected %v but got %v", expectedChecksum, gotchecksum)
+		} else {
+			gologger.Info().Msgf("Verified Integrity of %v", d.fullAssetName)
+		}
+	}
+
+	_ = UnpackAssetWithCallback(d.Format, bytes.NewReader(buff.Bytes()), getToolCallback)
+	return bin, errorutil.WrapfWithNil(err, "executable not found in archive") // Note: WrapfWithNil wraps msg if err != nil
+}
+
+// DownloadAssetWithName downloads asset with given name
+func (d *GHReleaseDownloader) DownloadAssetWithName(assetname string, showProgressBar bool) (*bytes.Buffer, error) {
+	assetID := 0
+	for _, v := range d.Latest.Assets {
+		if v.GetName() == assetname {
+			assetID = int(v.GetID())
+		}
+	}
+	if assetID == 0 {
+		return nil, errorutil.New("release asset %v not found", assetname)
+	}
+	resp, err := d.downloadAssetwithID(int64(assetID))
+	if err != nil {
+		return nil, errorutil.NewWithErr(err).Msgf("failed to download asset %v", assetname)
+	}
+	defer resp.Body.Close()
+
+	if showProgressBar {
+		bar := pb.New64(resp.ContentLength).SetMaxWidth(100)
+		bar.Start()
+		resp.Body = bar.NewProxyReader(resp.Body)
+		defer bar.Finish()
+	}
+
+	bin, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errorutil.NewWithErr(err).Msgf("failed to read resp body")
+	}
+	return bytes.NewBuffer(bin), nil
+}
+
+// DownloadSourceWithCallback downloads source code of latest release and calls callback for each file in archive
+func (d *GHReleaseDownloader) DownloadSourceWithCallback(showProgressBar bool, callback AssetFileCallback) error {
+	downloadURL := d.Latest.GetZipballURL()
+
+	resp, err := d.httpClient.Get(downloadURL)
+	if err != nil {
+		return errorutil.NewWithErr(err).Msgf("failed to source of %v", d.repoName)
+	}
+	defer resp.Body.Close()
+	if showProgressBar {
+		bar := pb.New64(resp.ContentLength).SetMaxWidth(100)
+		bar.Start()
+		resp.Body = bar.NewProxyReader(resp.Body)
+		defer bar.Finish()
+	}
+
+	bin, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errorutil.NewWithErr(err).Msgf("failed to read resp body")
+	}
+	return UnpackAssetWithCallback(Zip, bytes.NewReader(bin), callback)
 }
 
 // getLatestRelease returns latest release of error
-func (d *GHReleaseDownloader) GetLatestRelease() (*github.RepositoryRelease, error) {
-	release, resp, err := d.client.Repositories.GetLatestRelease(context.Background(), Organization, d.ToolName)
+func (d *GHReleaseDownloader) getLatestRelease() error {
+	release, resp, err := d.client.Repositories.GetLatestRelease(context.Background(), d.organization, d.repoName)
 	if err != nil {
+		errx := errorutil.NewWithErr(err)
 		if resp != nil && resp.StatusCode == 404 {
```

Comment:
Just a minor suggestion: what about using http.StatusCodeName instead of the number directly?

Reply:
done